app/controllers/blogs_controller.rb

```diff
@@ -11,7 +11,7 @@ def index
   end
 
   def show
-    raise ActiveRecord::RecordNotFound if @blog.secret && !@blog.owned_by?(current_user)
+    raise ActiveRecord::RecordNotFound if @blog.not_authorized?(current_user)
```

Comment:
```suggestion
    raise ActiveRecord::RecordNotFound unless @blog.authorized?(current_user)
```

否定形のメソッドよりも肯定形のメソッドの方が読みやすいかも？